src/state/config.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""haproxy-operator charm configuration."""
+
+import itertools
+import logging
+import typing
+
+# bandit flags import subprocess as a low security issue
+# We disable it here as it's simply a heads-up for potential issues
+from subprocess import STDOUT, check_output  # nosec B404
+
+import ops
+from pydantic import Field, ValidationError, field_validator
+from pydantic.dataclasses import dataclass
+
+logger = logging.getLogger()
+
+
+class InvalidCharmConfigError(Exception):
+    """Exception raised when a charm configuration is found to be invalid."""
+
+
+@dataclass(frozen=True)
+class CharmConfig:
+    """A component of charm state that contains the charm's configuration.
+
+    Attributes:
+        global_max_connection: The maximum per-process number of concurrent connections.
+        Must be between 0 and "fs.nr_open" sysctl config.
+    """
+
+    global_max_connection: int = Field(alias="global_max_connection")
+
+    @field_validator("global_max_connection")
+    @classmethod
+    def validate_global_max_conn(cls, global_max_connection: int) -> int:
+        """Validate global_max_connection config.
+
+        Args:
+            global_max_connection: The config to validate.
+
+        Raises:
+            ValueError: When the configured value is not between 0 and "fs.nr_open.
+
+        Returns:
+            int: The validated global_max_connection config.
+        """
+        # No user input so we're disabling bandit rule here as it's considered safe
+        output = check_output(  # nosec: B603
+            ["/usr/sbin/sysctl", "fs.nr_open"], stderr=STDOUT, universal_newlines=True
+        ).splitlines()
```

Comment:
Catch the `CalledProcessError` and throw a customer or validation exception here

Reply:
Thanks, I added handling for this exception.